gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -88,6 +95,11 @@
  * jobs. Upon completion of a job, it will either schedule the next job in the Dag (on SUCCESS) or mark the Dag as failed
  * (on FAILURE). Upon completion of a Dag execution, it will perform the required clean up actions.
  *
+ * For deleteSpec/cancellation requests for a flow URI, {@link DagManager} finds out the flowExecutionId using
+ * {@link JobStatusRetriever}, and forwards the request to the {@link DagManagerThread} which handled the addSpec request
+ * for this flow. We need separate {@value queue} and {@value cancelQueue} for each {@link DagManagerThread} because
+ * cancellation needs the information which is stored only in the same {@link Dag}.
```

Comment:
in the same {@link DagManagerThread} ? I am assuming you are referring to `dagsToJobs`

Reply:
yes, corrected.